optuna/storages/_base.py

```diff
@@ -458,6 +458,43 @@ def set_trial_values(self, trial_id: int, values: Sequence[float]) -> None:
         """
         raise NotImplementedError
 
+    # TODO(masap): When set_trial_state() and set_trial_values() are removed, make this function
+    # abstract and raise NotImplementedError.
+    def set_trial_state_values(
```

Comment:
Thanks for your review!

> So, could you change it as abstractmethod and remove set_trial_state() and set_trial_values() from BaseStorage?
> Also, in this case, we do not need to provide default implementation for this method.

According to comment from @toshihikoyanase, there are users who has user-defined storage classes.

 - https://github.com/optuna/optuna/pull/3272#issuecomment-1025398438
 - https://github.com/optuna/optuna/pull/3272#issuecomment-1025472081

Visible storage classes for users are `RDBStorage` and `RedisStorage`. So I guess users inherit them to create user-defined storage classes.

If we remove `set_trial_state()` and `set_trial_values()` and make `set_trial_state_values()` returns `NotImplementedError`, it breaks backward compatibility.

How do you think about this ?

Reply:
Thank you for the comment. We will discuss it and make decision for it.

Reply:
cc: @toshihikoyanase 
After the discussion, we reached the following consensus.

Currently, as we have not make BaseStorage public yet, in our policy it should be allowed to add breaking changes into BaseStorage.

Could you follow the first comment by me?

> Since we have not make BaseStorage public, we can change which methods are abstract and remove unnecessary methods. So, could you change it as abstractmethod and remove set_trial_state() and set_trial_values() from BaseStorage?

Reply:
> Currently, as we have not make BaseStorage public yet, in our policy it should be allowed to add breaking changes into BaseStorage.

Ok, It is reasonable. I removed the methods and test for user defined storage.